### PR TITLE
aws autoscaling api

### PR DESCRIPTION
**Pull request by nhibberd**
@markhibberd

### CODE REVIEW

**Comment by markhibberd** [comment]
Fair enough.

**Comment by markhibberd** [comment]
:+1:  But probably need to co-ordinate version conflict with Eduard's changes.

**Comment by nhibberd** [comment]
Also have another versioning issue at the baton levekw we should discuss tomorrow

File: amazonka-autoscaling/gen/Network/AWS/AutoScaling/Types.hs [hunk 1]
```diff
@@ -818,7 +818,7 @@ data AutoScalingGroup = AutoScalingGroup
     , _asgHealthCheckGracePeriod  :: Maybe Int
     , _asgHealthCheckType         :: Text
     , _asgInstances               :: List "member" Instance
-    , _asgLaunchConfigurationName :: Text
+    , _asgLaunchConfigurationName :: Maybe Text
```
Comment (charleso) [inline amazonka-autoscaling/gen/Network/AWS/AutoScaling/Types.hs]: Do we need to get a fix upstream as well? From a quick [glance](https://github.com/brendanhay/amazonka/blob/develop/amazonka-autoscaling/gen/Network/AWS/AutoScaling/Types/Product.hs#L223) `develop` doesn't have this change.

Comment (nhibberd) [inline amazonka-autoscaling/gen/Network/AWS/AutoScaling/Types.hs]: This goes all the way to the aws spec, can discuss today

**Comment by nhibberd** [comment]
Updated changes with issue #

**Pull request merged**